FirebaseAuth/Sources/Public/FirebaseAuth/FIRTOTPMultiFactorInfo.h

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#import <Foundation/Foundation.h>
+
+#import "FIRMultiFactorInfo.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+/** @class FIRTotpMultiFactorInfo
+    @brief Extends the MultiFactorInfo class for time based one-time password second factors.
+        The identifier of this second factor is "totp".
+        This class is available on iOS only.
+*/
+NS_SWIFT_NAME(TOTPMultiFactorInfo) API_UNAVAILABLE(macos, tvos, watchos)
+    @interface FIRTOTPMultiFactorInfo : FIRMultiFactorInfo
+
+/**
+   @brief This is the totp info for the second factor.
+*/
+@property(nonatomic, readonly, nullable) NSObject *totpInfo;
```

Comment:
What kind of information gonna be stored? Should developers access it?

Reply:
based on my research, only the base class MultiFactor is 100% necessary.  The only place I was planning on using this is to distinguish between the 2 MFA workflows we have right now when user clicks on the hints.  But we can use `factorID` to distinguish them and convert type later.  Please see the current sample App [here](https://github.com/firebase/firebase-ios-sdk/blob/master/FirebaseAuth/Tests/Sample/Sample/MainViewController%2BEmail.m#L136).

Moved it out of public folder.